pkg/cli/events.go

```diff
@@ -55,8 +61,22 @@ func (e *Events) Run(cmd *cobra.Command, args []string) error {
 	}
 
 	out := table.NewWriter(tables.Event, false, e.Output)
+	var flush bool
 	for event := range events {
-		out.Write(event)
+		if event != nil {
+			out.Write(event)
+		} else {
+			// A nil event means that it's time to flush
```

Comment:
This feels super kludgy.

Reply:
This does seem hacky. You might be abusing the table wrtier, but maybe it's fine.  I didn't know a tabwriter had a flush.  Fundamentally it has to buffer so that it can align columns so I don't understand how flush works.  Does it calculate column width historically for all rows, or just what is in the buffer.

Anyhow.  And alternate approach is that you just do a flush interval.  So one goroutine that a loop on an interval that calls flush. Like once a second.

Reply:
The behavior I've observed is that column width gets reset on flush, which makes for some pretty ugly tables.

Fortunately, it turns out that there's a drop-in replacement for `text/tabwriter` that can  remember column width https://pkg.go.dev/github.com/liggitt/tabwriter _and_ it's used by `kubectl` too.

Reply:
switched to that package and the formatting looks much better on repeated flush.